app/models/user_batch.rb

```diff
@@ -35,79 +36,247 @@ def create_users(mission)
     end
 
     # parse the input file as a spreadsheet
-    data = Roo::Spreadsheet.open(file)
+    @data = Roo::Spreadsheet.open(file)
 
+    validate_headers
+
+    if !@validation_error
+
+      # create the users in a transaction in case of validation error
+      User.transaction do
+
+        row_start = @data.first_row
+        @import_num = last_import_num_on_users
+
+        user_batch_attributes = parse_rows(row_start)
+
+        (0..number_of_iterations).each do |i|
+          current_attributes_batch = user_batch_attributes[row_start-1, BATCH_SIZE]
+
+          users_batch = create_users_instances(current_attributes_batch, mission)
```

Comment:
I'm a little concerned about creating 1000 ruby objects at once. I had envisioned this as populating and validating a single user object over and over. But if this performs well then ok.